extension/app/ts/backgroundServiceWorker.ts

```diff
@@ -28,3 +28,36 @@ const injectContentScript = async () => {
 }
 
 injectContentScript()
+
+/// Workaround for ChromiumIssue1316588 https://bugs.chromium.org/p/chromium/issues/detail?id=1316588
+// TODO, remove when chrome fixes the bug
+const storageArea = chrome.storage.local as chrome.storage.LocalStorageArea & { onChanged: chrome.storage.StorageChangedEvent }
+
+const TEST_INTERVAL_MS = 10000
+const STORAGE_WAIT_TIME_MS = 100
+
+const hasChromiumIssue1316588 = () => {
+	return new Promise((resolve) => {
+		let dispatched = false
+		const testEventDispatching = () => {
+			storageArea.onChanged.removeListener(testEventDispatching)
+			dispatched = true
+		}
+		storageArea.onChanged.addListener(testEventDispatching)
+		storageArea.set({ testEventDispatching: Math.random() })
+		setTimeout(() => resolve(!dispatched), STORAGE_WAIT_TIME_MS)
+	})
+}
+
+const fixChromiumIssue1316588 = () => {
+	hasChromiumIssue1316588().then((hasIssue) => {
+		if (hasIssue) {
+			chrome.runtime.reload()
+		} else {
+			setTimeout(fixChromiumIssue1316588, TEST_INTERVAL_MS)
+		}
+	})
+}
```

Comment:
Three suggestions here:
1. Use an async function instead of manual Promise stuff.
2. Use setInterval rather than queuing up a bunch of set timeouts in a row, if the intent is to do something every 10 seconds.
3. Don't silently swallow errors.
```suggestion
const intervalId = setInterval(fixChromiumIssue1316588, TEST_INTERVAL_MS)
const fixChromiumIssue1316588 = async () => {
	try {
		const hasIssue = await hasChromiumIssue1316588()
		if (!hasIssue) return
		chrome.runtime.reload()
		clearInterval(intervalId)
	} catch (error) {
		console.error(error)
	}
}
```